resources/lib/playerinfo.py

```diff
@@ -117,38 +125,45 @@ def onPlayBackEnded(self):  # pylint: disable=invalid-name
         self.quit.set()
         log(3, '[PlayerInfo %d] Event onPlayBackEnded' % self.thread_id)
         self.last_pos = self.total
-        # self.push_position(position=self.total, total=self.total, event='onPlayBackEnded')
 
     def onPlayBackError(self):  # pylint: disable=invalid-name
         ''' Called when playback stops due to an error '''
         if not self.listen:
             return
         self.quit.set()
         log(3, '[PlayerInfo %d] Event onPlayBackError' % self.thread_id)
-        # self.push_position(position=self.last_pos, total=self.total, event='onPlayBackError')
 
     def onPlayBackStopped(self):  # pylint: disable=invalid-name
         ''' Called when user stops Kodi playing a file '''
         if not self.listen:
             return
         self.quit.set()
         log(3, '[PlayerInfo %d] Event onPlayBackStopped' % self.thread_id)
-        # self.push_position(position=self.last_pos, total=self.total, event='onPlayBackStopped')
 
     def onThreadExit(self):  # pylint: disable=invalid-name
         ''' Called when player stops, before the player exited, so before the menu refresh '''
         log(3, '[PlayerInfo %d] Event onThreadExit' % self.thread_id)
-        self.push_position(position=self.last_pos, total=self.total, event='onThreadExit')
+        self.streampos = None
+        self.push_position(position=self.last_pos, total=self.total)
 
     def stream_position(self):
         ''' Get latest stream position while playing '''
-        self.push_upnext()
         while self.isPlaying() and not self.quit.is_set():
             self.update_position()
             if self.quit.wait(timeout=0.2):
                 break
         self.onThreadExit()
 
+    def add_upnext(self, video_id):
+        ''' Add Up Next url to Kodi Player '''
+        url = 'plugin://plugin.video.vrt.nu/play/upnext/%s' % video_id
+        if self.isPlaying() and self.getTotalTime() - self.getTime() < 1:
```

Comment:
Beware that because this is not atomic, either getTotalTime() or getTime() can cause a RuntimeError exception. I have seen this happen, which is why I wrote those wrappers.

Reply:
I'll use the wrappers.

Reply:
We could have better/other wrappers that return a value (or None) and handle the exception. 
The assignment can then go back in the real code. I think that will read better.